grails-app/domain/org/pih/warehouse/product/Product.groovy

```diff
@@ -457,12 +458,23 @@ class Product implements Comparable, Serializable {
     }
 
     /**
-     * Currently not implement since it would require coupling InventoryService to Product.
+     * Get quantity available to promise for this product given the location
      *
      * @param locationId
      */
-    def getQuantityAvailableToPromise(Integer locationId) {
-        throw new NotImplementedException()
+    def getQuantityAvailableToPromise(String locationId) {
+        def productAvailability = ProductAvailability.createCriteria().list {
+            resultTransformer(Criteria.ALIAS_TO_ENTITY_MAP)
+            projections {
+                sum("quantityAvailableToPromise", "quantityAvailableToPromise")
+            }
+            location {
+                eq("id", locationId)
+            }
+            eq("product", this)
+        }
+
+        return productAvailability?.get(0)?.quantityAvailableToPromise ?: 0
```

Comment:
The .get(0) might throw an exception if there are no records in the table, so you should consider handling that edge case.